pkg/auth/signing/encode.go

```diff
@@ -32,3 +32,11 @@ func EncodePieceHash(hash *pb.PieceHash) ([]byte, error) {
 	defer func() { hash.Signature = signature }()
 	return proto.Marshal(hash)
 }
+
+// EncodeVoucher encodes voucher into bytes for signing.
+func EncodeVoucher(voucher *pb.Voucher) ([]byte, error) {
```

Comment:
You are just serializing the bytes. Not quite encoding. renaming as "SerializeVoucher" or something in those line may be more appropriate